src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -879,3 +907,132 @@ extern "C" UINT __stdcall ExecNetFx(
     return WcaFinalize(er);
 }
 
+/******************************************************************
+ DotNetCompatibilityCheck - entry point for NetFx Custom Action
+
+*******************************************************************/
+extern "C" UINT __stdcall DotNetCompatibilityCheck(
+    __in MSIHANDLE hInstall
+    )
+{
+//    AssertSz(FALSE, "debug DotNetCompatibilityCheck");
+
+    HRESULT hr = S_OK;
+    UINT er = ERROR_SUCCESS;
+
+    DWORD dwTempPathLength = 0;
+    LPWSTR pwzTempPath = NULL;
+    PMSIHANDLE hView = NULL;
+    PMSIHANDLE hRec = NULL;
+    LPWSTR pwzPlatform = NULL;
+    LPWSTR pwzNetCoreCheckBinaryId = NULL;
+    LPWSTR pwzNetCoreCheckFilePath = NULL;
+    LPWSTR pwzRuntimeType = NULL;
+    LPWSTR pwzVersion = NULL;
+    LPWSTR pwzRollForward = NULL;
+    LPWSTR pwzVariable = NULL;
+    LPWSTR pwzCommandLine = NULL;
+
+    // Initialize
+    hr = WcaInitialize(hInstall, "DotNetCompatibilityCheck");
+    ExitOnFailure(hr, "failed to initialize");
+
+    // Get temp directory path
+    dwTempPathLength = GetTempPathW(dwTempPathLength, NULL);
```

Comment:
Use `PathCreateTempFile()` from `pathutil.h`. It handles the string allocation for you. Pass `NULL` for `phTempFile` if you don't need the file handle.

Reply:
Thinking about this more, I think we actually need to extract the exe to a randomly generated folder name. Then an attacker cannot attempt DLL hijack by simply dropping a file in `%TMP%`. `PathCreateTempDirectory()` with a random string should work.

Reply:
This is still an issue I called out in my last review.

Reply:
You're talking about [this comment](https://github.com/wixtoolset/wix4/pull/262#discussion_r1003656622) right? I was just waiting for confirmation on what to do there (whether to optimize with pre-exporting or not), I'll make the change to export binary to a random directory and push those changes.